compiler/noirc_frontend/src/hir/comptime/interpreter/builtin.rs

```diff
@@ -675,15 +672,24 @@ fn quoted_as_expr(
 ) -> IResult<Value> {
     let argument = check_one_argument(arguments, location)?;
 
-    let expr_parser = parser::expression().map(|expr| Value::expression(expr.kind));
-    let statement_parser = parser::fresh_statement().map(Value::statement);
-    let lvalue_parser = parser::lvalue(parser::expression()).map(Value::lvalue);
-    let parser = choice((expr_parser, statement_parser, lvalue_parser));
-    let parser = parser.then_ignore(just(Token::Semicolon).or_not());
+    let result =
+        parse(interner, argument.clone(), Parser::parse_expression_or_error, "an expression");
+    if let Ok(expr) = result {
+        return option(return_type, Some(Value::expression(expr.kind)));
+    }
 
-    let expr = parse(interner, argument, parser, "an expression").ok();
+    let result =
+        parse(interner, argument.clone(), Parser::parse_statement_or_error, "an expression");
+    if let Ok(stmt) = result {
+        return option(return_type, Some(Value::statement(stmt.kind)));
+    }
```

Comment:
This reminded me that we could have a `parse_statement_or_expression_or_lvalue` that returns either of those in one go. I think it's doable. That would mean no need to `argument.clone()` and it'll be slightly faster, which is important for macro code. But maybe this could be for a follow-up PR... for now I'll try to reduce the repetition.

Reply:
I couldn't find a quick way to reduce this repetition, but... it's the only repetition, and it's a small one. Once we have `parse_statement_or_expression_or_lvalue` it'll be gone 🤞